sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -4,54 +4,244 @@
 #nullable enable
 
 using System;
+using System.Collections;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Linq.Expressions;
 using System.Reflection;
 
 namespace Azure.Core.Pipeline
 {
+#pragma warning disable CA1001 // Implement IDisposable
     internal readonly struct DiagnosticScope : IDisposable
+#pragma warning restore CA1001 // Implement IDisposable
     {
-        private readonly DiagnosticActivity? _activity;
+        private static readonly ConcurrentDictionary<string, object?> ActivitySources = new();
 
-        private readonly string _name;
+        private readonly ActivityAdapter? _activityAdapter;
 
-        private readonly DiagnosticListener _source;
+        internal DiagnosticScope(string ns, string scopeName, DiagnosticListener source, ActivityKind kind)
```

Comment:
Given library owners will call this, spelling the attributes out (I'm assuming "namespace" here) would be nice, if not XML doc comments for all publics and internals for shared source.

Reply:
This method is not supposed to be called directly, library owners are calling `ClientDiagnostics.CreateScope`